app/controllers/api/v1/product_share_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::ProductShareController < Api::V1::BaseController
+    def create
+        product_share_email
+    end
+
+    def product_share_email
+        ProductMailer
+          .with(
+            to_email: share_params[:to_email],
+            product_link: share_params[:product_link],
+            product_name: share_params[:product_name],
+            product_price: share_params[:product_price]
+          )
+          .share_email
+          .deliver_later
+    end
+
+    private
+
+    def share_params
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/IndentationWidth: Use 2 (not 4) spaces for indentation.